datastream-server/src/main/java/com/linkedin/datastream/server/AssignmentTaskMapLogger.java

```diff
@@ -0,0 +1,84 @@
+/**
+ *  Copyright 2023 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import org.slf4j.Logger;
+
+import com.linkedin.datastream.common.zk.ZkClient;
+
+
+/**
+ * Class that logs assignments one task at a time, keeping each log line under 1 MB in size
+ */
+public class AssignmentTaskMapLogger {
+  private final Logger _log;
+  private static final double ONE_MEBIBYTE = 1024 * 1024;
+  private static final double ONE_MB = 1.0;
+  private static final double BUFFER_1KB =  1.0 / 1024.0;
+  private static final int BUFFER_ADJUSTED_1MB = 1024 * 1020;
+
+  /**
+   * Constructor
+   * @param logger logger to be used for logging
+   */
+  public AssignmentTaskMapLogger(Logger logger) {
+    _log = logger;
+  }
+
+  /**
+   * logs assigned tasks for each host
+   */
+  public void logAssignment(Map<String, Set<DatastreamTask>> taskMap) {
+    for (String host: taskMap.keySet()) {
+      for (DatastreamTask task: taskMap.get(host)) {
+        logTask(task.toString(), host, 1);
+      }
+    }
+  }
+
+  /**
+   * prints one log line for each task less than 1MB in size
+   * @param task
+   */
+  private void logTask(String task, String host, int part) {
```

Comment:
instead of using this only for task map, lets make this as a logger utility which can be used for parameterized string with parameterized limit.

and lets move it to datastream-utils/src/main/java/com/linkedin/datastream/common/LogUtils.java ?